test/pylib/runner.py

```
@@ -316,13 +311,60 @@ def testpy_cluster_factory(request: pytest.FixtureRequest,
+
+            artifacts.add_exit_artifact(uninstall)
+
+        await cluster.install_and_start()
+        # If cluster failed to start, raise the exception immediately
+        # so a broken cluster is never handed out to tests
+        if cluster.start_exception is not None:
+            # Clean up the broken cluster before raising
+            try:
+                await cluster.recycle()
+            except:
+                pass  # Ignore cleanup errors
```

Ruff (0.16.1)

[warning] Use `contextlib.suppress(BaseException)` instead of `try`-`except`-`pass`

(SIM105)